### PR TITLE
Fix disable validation

### LINKED ISSUE #83: Training without validation set crashes
At the moment, the training crashes when ``DictModule`` does not split the dataset into training and validation because ``Lightning`` always calls ``val_dataloader()``.

### PR DESCRIPTION
**Pull request by luigibonati**
## Description
This PR  fixes #83 

## Todos
I changed the `DictModule` as to accept a list with just one split. Furthermore, since by default the trainer expects a validation loader, it has to be disabled in the trainer options (see https://github.com/Lightning-AI/lightning/issues/12374) 
I added these instructions in the tutorial on how to customize training

## Status
- [ ] Ready to go

### CODE REVIEW

**Review by andrrizzi — APPROVED** [review]
Looks good to me!

**Comment by luigibonati** [comment]
I did not merge this yet as I realized there is an issue with the logging, in that the `MetricsCallback` that I wrote is called only at the end of validation, so it does not log the training values. needs to fix that before merging

**Comment by luigibonati** [comment]
I fixed the issue by changing the moment in which the callback is called (at the end of the training and not validation), merging it now

**Pull request merged**
